packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L1StandardBridge.sol

```diff
@@ -0,0 +1,314 @@
+     * @param _amount Amount of the ERC20 to deposit.
+     * @param _l2Gas Gas limit required to complete the deposit on L2.
+     * @param _data Optional data to forward to L2. This data is provided
+     *        solely as a convenience for external contracts. Aside from enforcing a maximum
+     *        length, these contracts provide no guarantees about its content.
+     */
+    function _initiateERC20Deposit(
+        address _l1Token,
+        address _l2Token,
+        address _from,
+        address _to,
+        uint256 _amount,
+        uint32 _l2Gas,
+        bytes calldata _data
+    )
+        internal
+    {
+        // When a deposit is initiated on L1, the L1 Bridge transfers the funds to itself for future withdrawals.
+        IERC20(_l1Token).safeTransferFrom(
+            _from,
+            address(this),
+            _amount
+        );
+
+        // Construct calldata for _l2Token.finalizeDeposit(_to, _amount)
+        bytes memory message = abi.encodeWithSelector(
+            iOVM_L2ERC20Bridge.finalizeDeposit.selector,
+            _l1Token,
+            _l2Token,
+            _from,
+            _to,
+            _amount,
+            _data
+        );
+
+        // Send calldata into L2
+        sendCrossDomainMessage(
+            l2TokenBridge,
+            _l2Gas,
+            message
+        );
+
+        deposits[_l1Token][_l2Token] = deposits[_l1Token][_l2Token].add(_amount);
+
+        // We omit _data here because events only support bytes32 types.
+        emit ERC20DepositInitiated(_l1Token, _l2Token, _from, _to, _amount, _data);
+    }
+
+    /*************************
+     * Cross-chain Functions *
+     *************************/
+
+     /**
+     * @inheritdoc iOVM_L1StandardBridge
+     */
+    function finalizeETHWithdrawal(
+        address _from,
+        address _to,
+        uint256 _amount,
+        bytes calldata _data
+    )
+        external
+        override
+        onlyFromCrossDomainAccount(l2TokenBridge)
+    {
+        (bool success, ) = _to.call{value: _amount}(new bytes(0));
+        require(success, 'TransferHelper::safeTransferETH: ETH transfer failed');
+
+        emit ETHWithdrawalFinalized(_from, _to, _amount, _data);
+    }
+
+    /**
+     * @inheritdoc iOVM_L1ERC20Bridge
+     */
+    function finalizeERC20Withdrawal(
+        address _l1Token,
+        address _l2Token,
+        address _from,
+        address _to,
+        uint256 _amount,
+        bytes calldata _data
+    )
+        external
+        override
+        onlyFromCrossDomainAccount(l2TokenBridge)
+    {
+        deposits[_l1Token][_l2Token] = deposits[_l1Token][_l2Token].sub(_amount);
+
+        // When a withdrawal is finalized on L1, the L1 Bridge transfers the funds to the withdrawer.
+        IERC20(_l1Token).safeTransfer(_to, _amount);
+
+        emit ERC20WithdrawalFinalized(_l1Token, _l2Token, _from, _to, _amount, _data);
+    }
+
+    /*****************************
+     * Temporary - Migrating ETH *
```

Comment:
General nit for this file to remove these blocks and simply replace with "public functions", "internal functions", "constructor", etc.

Reply:
We inherited this styling from before so we just kept it. Also I'm not a fan of these blocks separated by function visibility